### ISSUE TITLE
RequestQueue.cancelAll doesn't use Object.equals

### ISSUE THREAD

**Issue by CapnSpellcheck** [google/volley#152]
```
public boolean apply(Request<?> request) {
                return request.getTag() == tag;
            }

```

Why shouldn't it use Object.equals?

**Comment by jpd236** [comment]
That's a very good question. Unfortunately it appears to have always been this way, and is even documented as such in the cancelAll() Javadoc, which means it's not something we can change without breaking the API contract.

The original change adding this was at https://android.googlesource.com/platform/frameworks/support/+/19a56e8495477bfd89e4c4913860fdbcb7beb504, and even internally I don't see any explanation as to why it was done this way.

We can look at adding a new API and deprecating the old one in 1.2.0. In the mean time, it's possibly to specify your own RequestFilter as follows:
```java
cancelAll(new RequestFilter() {
    @Override
    public boolean apply(Request<?> request) {
        return Objects.equals(tag, request.getTag());
    }
});
```

**Comment by CapnSpellcheck** [comment]
Of course. I just wanted to point this out :)

**Comment by runforprogram** [comment]
I  think it is a feature, _RequestManager.cancelAll_ is a danger method which will to cause all of requests which tag equaled  be canceled. use  _==_  can be minimize side effect.think this situation  has two place  request a same url and use  a object like a url _String_
,want cancel A request but not B, use _==_ will minimal probability cause B canceled than use 
```java
 Objects.equals(tag, request.getTag())
```

**Comment by joimxjtuse** [comment]
i agree with  @runforprogram, compare  '==' with String.equals(),   just for efficiency.

**Comment by joebowbeer** [comment]
In Request, the tag is defined as an `opaque` token.

A unit test for this functionality:

[RequestQueueTest.java#L52](https://github.com/google/volley/blob/master/src/test/java/com/android/volley/RequestQueueTest.java#L52)
